### PR TITLE
SamplerV2 update

### LINKED ISSUE #1313: Primitive interface updates for SamplerV2
Part of https://github.com/Qiskit/qiskit-ibm-runtime/issues/1131 

Support SamplerV2 input/output:
- [RFC](https://github.com/Qiskit/RFCs/pull/56)
- Input/output would also be PUB based
- Output is per shot measurement, with convenience functions to returns counts/probabilities

Based on the following Qiskit PRs https://github.com/Qiskit/qiskit/pull/11566 (replaced https://github.com/Qiskit/qiskit/pull/11264)
- SamplerV2 Container https://github.com/Qiskit/qiskit/pull/11542
- BaseSamplerV2 https://github.com/Qiskit/qiskit/pull/11529

Work should be done off the experimental [branch](https://github.com/Qiskit/qiskit-ibm-runtime/tree/experimental-0.2) where some of the work is already done (https://github.com/Qiskit/qiskit-ibm-runtime/pull/1224)

### PR DESCRIPTION
**Pull request by gadial**
### Summary
Finalizes the `SamplerV2` class and adds relevant tests.

Replaces #1326 

### Details and comments
Fixes #1313 

Most of the work required to run `SamplerV2` was done in previous PR's and support for V2 containers in `qiskit` is now full, so this PR mainly adds `SamplerV2` to the Primitives V2 unit tests, and copies over the `SamplerV2` integration tests from qiskit.

### CODE REVIEW

**Comment by mberna** [comment]
Converting to draft as per @gadial's request.

File: qiskit_ibm_runtime/sampler.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ class Sampler:
     version = 0
 
 
-class SamplerV2(BasePrimitiveV2[SamplerOptions], Sampler, BaseSamplerV2):
+class SamplerV2(BasePrimitiveV2, Sampler, BaseSamplerV2):
```
Comment (jyu00) [inline qiskit_ibm_runtime/sampler.py:45]: Why was `SamplerOptions` removed? It's meant to set the expected options class.

File: qiskit_ibm_runtime/sampler.py [hunk 2]
```diff
@@ -148,23 +150,16 @@ class SamplerV1(BasePrimitiveV1, Sampler, BaseSampler):
 
     Example::
 
-        from qiskit.circuit import QuantumCircuit, QuantumRegister, ClassicalRegister
+        from qiskit.test.reference_circuits import ReferenceCircuits
```
Comment (jyu00) [inline qiskit_ibm_runtime/sampler.py:153]: `qiskit.test` no longer exists in 1.0

File: qiskit_ibm_runtime/sampler.py (hunk 1 above)
Comment (gadial) [reply qiskit_ibm_runtime/sampler.py:45]: Seems to me I accidently reverted to an older version. Will fix.

File: qiskit_ibm_runtime/sampler.py (hunk 2 above)
Comment (gadial) [reply qiskit_ibm_runtime/sampler.py]: Replaced with explicit code for generating bell circuit - is there a better alternative? Also, we have similar example in `provider_session.py`

**Review by jyu00 — APPROVED** [review]

**Pull request merged**